hexgame/src/board.py

```diff
@@ -103,17 +104,72 @@ def find_neighbours(self, coords: tuple[int,int]) -> set[Cell]:
                 else:
                     #left border inside
                     return set((Cell(0,y-1),Cell(0,y+1),Cell(1,y),Cell(1,y+1)))
-
             elif x == self.dim_x - 1:
+                #right border
                 if y == 0:
                     #bottom right
                     return set((Cell(self.dim_x - 2,0),Cell(self.dim_x - 1,1)))
                 elif y == (self.dim_y -1):
                     #top right
-                    return set((Cell(self.dim_x - 2,self.dim_y-2),
-                    Cell(self.dim_x - 2,self.dim_y-1),
-                    Cell(self.dim_x - 1,self.dim_y-2)))            
+                    return set(
+                        (
+                            Cell(self.dim_x - 2,self.dim_y-2),
+                            Cell(self.dim_x - 2,self.dim_y-1),
+                            Cell(self.dim_x - 1,self.dim_y-2)
+                        )
+                    )  
+                else:
+                    #right border inside
+                    return set(
+                        (
+                            Cell(self.dim_x - 2,self.dim_y-2),
+                            Cell(self.dim_x - 2,self.dim_y-1),
+                            Cell(self.dim_x - 1,self.dim_y-2),
+                            Cell(self.dim_x - 1,self.dim_y-2),
+                        )
+                    )
+            elif y == 0:
```

Comment:
I don't like that we are introducing all of this redundant code.
Also the function became very long, can we keep it shorter ?
I am thinking that also we are not considering when we are searching a neighbour in a very small board that will not have all the tiles around the target coord in the same way it happens from the 3 by 3 case and bigger boards.
I suggest that we check that all the things that we are returning in the set are part of the board  ( there is a function that tells us that `has_cell()` ).
For the if else conditions, we can say that there are only up to 9 different cases possible.
Therefore I would do:
if x ==0 and y==0 :   # bottom left
elif  x==0 and y==max-1 :  # top left
...
for the 8 cases in the border and then 
else :    # case in the middle

In this way we avoid nested code  as much as possible.
What do you think ?

Reply:
Yes, very fair.

As I writing and copy pasting I was telling to myself I just wanted to close the pr and get something working so that for example we could have added test and then refactor later , but I guess there is no reason as why that cannot be done all in once now.

And ah, that makes sense then as why you made that has cell. I also had thought of making it mushc simpler by creating the nbd just by adding a neighbour cell provided it exist.

Reply:
Done this now, I have: rewritten the unit test to be much simple using fixtures.
Then I have reimplemented the find neighbours functions in a much easier way, test all pass.
Along the way I also aoutformatted with pep8, sorry about that we said we were going to do in another PR